scripts/git-hooks/pre-commit

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+FAILED=0
+
+# Colors for output
+BOLD="\033[1m"
+GREEN="\033[1;32m"
+RED="\033[1;31m"
+RESET="\033[0m"
+
+echo "${BOLD}Running pre-commit hooks...${RESET}"
+
+# Check if there are staged changes in Python files
+PYTHON_STAGED_FILES=$(git diff --name-only --diff-filter=ACM | grep -E "\.py$")
+if [ -n "$PYTHON_STAGED_FILES" ]; then
+  echo "${BOLD}Running backend linters...${RESET}"
+  make lint
+  BACKEND_RESULT=$?
+  if [ $BACKEND_RESULT -ne 0 ]; then
+    echo "${RED}Backend linting failed!${RESET}"
+    echo "${RED}Please fix the issues and try committing again.${RESET}"
+    FAILED=1
+  else
+    echo "${GREEN}Backend linting passed!${RESET}"
+  fi
+else
+    echo "${BOLD}No Python files staged for commit. Skipping backend linters...${RESET}"
+fi
+
+# Check if there are staged changes in frontend files
+FRONT_STAGED_FILES=$(git diff --name-only --diff-filter=ACM | grep -E "src/frontend/.*\.(js|jsx|ts|tsx)$")
+if [ -n "$FRONT_STAGED_FILES" ]; then
+  echo "${BOLD}Running frontend linters...${RESET}"
+  make lint-front
```

Comment:
Unlike the backend linter, the frontend linter doesn't write its improvements, this should be the default for frontend linting as well.

Reply:
This is out of scope for this PR, but feel free to add commits on this if you see things which can be improved of course